src/lexers/jsx-lexer.js

```diff
@@ -36,7 +35,13 @@ export default class JsxLexer extends JavascriptLexer {
       ...JSXParserExtension
     })
 
-    this.acorn = injectAcornJsx(this.acorn)
+    try {
+      const injectAcornJsx = require('acorn-jsx/inject')
```

Comment:
This is my current approach for requiring optional dependencies. There are [packages](https://github.com/jchip/optional-require) that would allow us to do this with a bit less code, though I suppose those would still require a try/catch.

Does it make sense to have this logic in the constructors?

Reply:
I'm fine with this!